lib/rucio/web/rest/flaskapi/v1/requests.py

```diff
@@ -27,7 +27,7 @@
 from rucio.web.rest.flaskapi.v1.common import ErrorHandlingMethodView, check_accept_header_wrapper_flask, generate_http_error_flask, parse_scope_name, response_headers, try_stream
 
 if TYPE_CHECKING:
-    from typing import Iterator
+    from collections.abc import Iterator
```

Comment:
I think we should recommend that modules from the Python standard library are not conditionally imported, even if used exclusively for type hints.

Reply:
`flake8-type-checking` has a setting to achieve this: https://docs.astral.sh/ruff/settings/#lintflake8-type-checking

I opened this issue to track this work: https://github.com/rucio/rucio/issues/6728

Reply:
On the other hand, this page makes a good point (https://docs.astral.sh/ruff/rules/typing-only-standard-library-import/):

> Unused imports add a performance overhead at runtime, and risk creating import cycles. If an import is only used in typing-only contexts, it can instead be imported conditionally under an if TYPE_CHECKING: block to minimize runtime overhead.

And there's this setting (https://docs.astral.sh/ruff/settings/#lint_flake8-type-checking_quote-annotations) which enables this:

> Whether to add quotes around type annotations, if doing so would allow the corresponding import to be moved into a type-checking block.

I think it might be worth following these guidelines and enabling the linter to check for imports that are only used in a type-checking context, and place them in an `IF TYPE_CHECKING` block to lower runtime overhead. What do you think?